src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Location/ShowLocationHistory/ShowLocationHistory.xaml.cs

```diff
@@ -0,0 +1,153 @@
+            _polylineBuilder = new PolylineBuilder(SpatialReferences.Wgs84);
+
+            // Start location services.
+            HandleLocationReady();
+        }
+
+        private async void HandleLocationReady()
+        {
+            // Create the location data source.
+            _locationDataSource = new FakeLocationDataSource();
+            // Use this instead if you want real location: _locationDataSource = new SystemLocationDataSource();
+
+            try
+            {
+                // Start the data source.
+                await _locationDataSource.StartAsync();
+
+                if (_locationDataSource.IsStarted)
+                {
+                    // Set the location display data source and enable location display.
+                    MyMapView.LocationDisplay.DataSource = _locationDataSource;
+                    MyMapView.LocationDisplay.IsEnabled = true;
+                    MyMapView.LocationDisplay.AutoPanMode = LocationDisplayAutoPanMode.Recenter;
+                    MyMapView.LocationDisplay.InitialZoomScale = 10000;
+
+                    // Enable the button to start location tracking.
+                    LocationTrackingButton.IsEnabled = true;
+                }
+                else
+                {
+                    ShowMessage("There was a problem enabling location", "Error");
+                }
+            }
+            catch (Exception e)
+            {
+                System.Diagnostics.Debug.WriteLine(e);
+                ShowMessage("There was a problem enabling location", "Error");
+            }
+        }
+
+        private void ToggleLocationTracking()
+        {
+            // Toggle location tracking.
+            _isTrackingEnabled = !_isTrackingEnabled;
+
+            // Apply new configuration.
+            if (_isTrackingEnabled)
+            {
+                // Configure new symbology first.
+                _locationDataSource.LocationChanged += LocationDataSourceOnLocationChanged;
+
+                // Update the UI.
+                LocationTrackingButton.Content = "Stop tracking";
+            }
+            else
+            {
+                // Stop updating.
+                _locationDataSource.LocationChanged -= LocationDataSourceOnLocationChanged;
+
+                // Update the UI.
+                LocationTrackingButton.Content = "Start tracking";
+            }
+        }
+
+        private void LocationDataSourceOnLocationChanged(object sender, Location e)
+        {
+            // Add the location to the history.
+            _polylineBuilder.AddPoint(e.Position);
+
+            // Show the location point.
+            _locationHistoryOverlay.Graphics.Add(new Graphic(e.Position));
+
+            // Remove the old line.
+            _locationHistoryLineOverlay.Graphics.Clear();
+
+            // Add the updated line.
+            _locationHistoryLineOverlay.Graphics.Add(new Graphic(_polylineBuilder.ToGeometry()));
+        }
```

Comment:
I don't like the way the "historical" points jump ahead of the current location marker. I'm not sure if this is a good way to illustrate it (might overcomplicate or confuse), but the code below will show a more traditional "tracking" behind the animated current position.
```csharp
private MapPoint _lastPosition = null;
private void LocationDataSourceOnLocationChanged(object sender, Location e)
{
    // Remove the old line.
    _locationHistoryLineOverlay.Graphics.Clear();

    // Add any previous position to the history.
    if(_lastPosition != null)
    {
        _polylineBuilder.AddPoint(_lastPosition);
        _locationHistoryOverlay.Graphics.Add(new Graphic(_lastPosition));
    }
            
    // Store the current position.
    _lastPosition = e.Position;

    // Add the updated line.
    _locationHistoryLineOverlay.Graphics.Add(new Graphic(_polylineBuilder.ToGeometry()));
}
```